src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2157,6 +2164,21 @@ private static void AddEntryPointCandidates(
             ArrayBuilder<MethodSymbol> entryPointCandidates, IEnumerable<Symbol> members)
         {
             foreach (var member in members)
+            {
+                if (member.GetIsNewExtensionMember())
+                {
+                    if (member is MethodSymbol method && method.TryGetCorrespondingExtensionImplementationMethod() is { } implementationMethod)
+                    {
+                        addIfCandidate(entryPointCandidates, implementationMethod);
```

Comment:
>addIfCandidate(entryPointCandidates, implementationMethod); [](http://example.com/codeflow?start=24&length=59)

It is not obvious why we need to do this. Wouldn't the implementation methods be enumerated as members of the enclosing class?

Reply:
The search for entry point uses `GetSymbolsWithNameCore` which relies on `NameSymbolSearcher`/`AbstractSymbolSearcher` which walks through the declaration tree (which doesn't include the implementation methods)